core/src/main/scala/chisel3/experimental/HasTypeAlias.scala

```diff
@@ -0,0 +1,51 @@
+package chisel3.experimental
+
+import chisel3.Record
+
+/** Wrapper object for a Record alias name. Primarily intended to provide an invocation point for source line locators, but
+  * also contains pertinent information to generating FIRRTL alias statements.
+  *
+  * @param id The desired name to generate an alias statement
+  * @param strippedSuffix In the case of forced coersion by [[Input]] or [[Output]], the string to append to the end of the
+  *        alias name. Takes the default value of `"_stripped"`
+  */
+case class BundleAlias private[chisel3] (info: SourceInfo, id: String, strippedSuffix: String = "_stripped")
+
+object BundleAlias {
+  def apply(id: String)(implicit info:  SourceInfo): BundleAlias = BundleAlias(info, id)
+  def apply(id: String, strippedSuffix: String)(implicit info: SourceInfo): BundleAlias =
+    BundleAlias(info, id, strippedSuffix)
+}
+
+trait HasTypeAlias {
+  this: Record =>
+
+  /** An optional FIRRTL type alias name to give to this [[Record]]. If overrided with a `Some(...)`, for instance
+    * `Some(BundleAlias("UserBundle"))`, this causes emission of circuit-level FIRRTL statements that declare that name for
+    * this bundle type:
+    *
+    * @example
+    * {{{
+    * class MyBundle extends Bundle with HasTypeAlias {
+    *   override def aliasName = Some("UserBundle")
```

Comment:
isn't this wrong? Don't they need to create a Some(BundleAlias(...))

Could we turn this example into a really quick cookbook in the naming cookbook?

Reply:
maybe right after here: https://github.com/chipsalliance/chisel/blob/main/docs/src/cookbooks/naming.md#how-would-i-write-my-own-typename-for-my-data-types